src/audio/mux/mux.c

```diff
@@ -426,7 +426,7 @@ static int demux_copy(struct comp_dev *dev)
 		sink = container_of(clist, struct comp_buffer, source_list);
 		sink_c = buffer_acquire(sink);
 
-		if (sink_c->sink->state == dev->state) {
+		if (sink_c->sink && sink_c->sink->state == dev->state) {
```

Comment:
and here each individual case has to be considered... But again - can NULL actually occur in each of those cases? If yes - shouldn't the source of the problem be fixed? Maybe we should drop all these NULL source / sink changes and only fix them *if and when* we hit those exceptions

Reply:
oh man we are so running in circles...

The sink_c->sink can be invalid some some components, we shouldn't have to think on every case and check if this can happen or not. I completely disagree with a one-at-a-time fix if and when we hit a firmware oops due to an invalid pointer dereference.

At this point my recommendation is: fix the buffer acquire infrastructure and account for all possible cases in the framework, not in users of that framework.

@lyakh @lgirdwood back to your court.